pkg/api/types.go

```diff
@@ -2570,6 +2570,25 @@ type RangeAllocation struct {
 	Data []byte `json:"data"`
 }
 
+// APIServerIPInfo is an opaque API object (not exposed to end users) that is temporarily persisted (with
+// a short TTL) to etcd to indicate that a particular IP is a viable API server.  The list of such objects
+// is used to determine the endpoints for the autogenerated "kubernetes" service.
+type APIServerIPInfo struct {
```

Comment:
I think we want separate objects and keys anyway.  However, we don't need a new API type.  Just store one endpoints object per apiserver.